sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2435,17 +2450,6 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
-  val LEGACY_ALLOW_DUPLICATED_MAP_KEY =
```

Comment:
We need to update sql-migration-guide doc too.

Reply:
+1,
Also find another mistake I made in the migration guide, fixed in #27782, please take a look.